### PR TITLE
WNMGDS-1204 dialog icon prop

### PR DESCRIPTION
**Pull request by scheul93**
## Summary
Updating the Dialog component to have an icon prop that allows for a custom close icon

### Added
* `closeIcon` prop for Dialog component that is defaulted to the <CloseIcon />
* In medicare, defaulted the new prop to <CloseIconThin />

### Removed
* Dialog close icon background-image

### Changed
* Dialog documentation HTML examples to reflect the changes

## How to test
Run `yarn start` to build the core doc site. Compare [local Modal Dialog](http://localhost:3000/components/dialog/) page to [production](https://design.cms.gov/components/dialog/). Confirm the dialog close buttons look the same

Run `yarn start:medicare` to build the mgov doc site. Compare to [prod site](https://cmsgov.github.io/design-system/medicare/components/dialog/). Confirm close button looks the same.

@ericwbailey, I ran Axe and didn't see any errors with updating the close button like this, but please let me know if anything stand out to you accessibility-wise.

### CODE REVIEW

**Comment by forrestbaer** [comment]
The close icon in the react dialog example is bigger for me, also it seems pretty far to the left.

image

**Comment by scheul93** [comment]
> The medicare close icon in the varying size dialog examples is bigger for me, also it seems pretty far to the left.
Oh! Your screenshot looks like it from the size example which I did not overwrite in mgov. I will look into it and update!

File: packages/design-system-docs/src/pages/components/Dialog/dialog-size.example.html [hunk 1]
```diff
@@ -30,9 +30,26 @@
       <header class="ds-c-dialog__header" role="banner">
         <h1 class="ds-h2" id="dialog-heading">Dialog heading</h1>
         <button
-          class="dialog-close-btn ds-c-button ds-c-button--transparent ds-c-dialog__close"
```
Comment (pwolfert) [inline packages/design-system-docs/src/pages/components/Dialog/dialog-size.example.html:33]: I agree that this should probably go, but it's referenced by some commented out JavaScript at the bottom of this file. I don't know the origin of the code, but I imagine it's helpful to show people how you make it interactive?

**Review by pwolfert — COMMENTED** [review]
Looks like this still needs to update from master in order to build, but the code is looking good. One comment so far

File: packages/design-system-docs/src/pages/components/Dialog/dialog-size.example.html (hunk 1 above)
Comment (scheul93) [reply packages/design-system-docs/src/pages/components/Dialog/dialog-size.example.html:33]: I added it back!

**Review by forrestbaer — APPROVED** [review]
👍

File: packages/ds-medicare-gov/src/components/Dialog/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Dialog, CloseIconThin } from '@cmsgov/design-system';
+
+Dialog.defaultProps = {
+  ...Dialog.defaultProps,
+  closeButtonText: null,
```
Comment (pwolfert) [inline packages/ds-medicare-gov/src/components/Dialog/index.js:6]: I've always thought this is kind of an accessibility issue 😕. At least it's really obvious that this feature is being removed, so we can address it in the future.

**Review by pwolfert — APPROVED** [review]
LGTM 👍

**Pull request merged**